File: leetcode/leetcode1346.py
import collections
from typing import List


class Solution:
    def checkIfExist(self, arr: List[int]) -> bool:
        m = collections.Counter(arr)
        for n in arr:
            if n == 0 and m[n] >= 2:
                return True
            if n != 0 and m[n * 2] >= 1:
                return True
        return False

File: leetcode/leetcode1346.cpp
class Solution {
public:
    bool checkIfExist(vector<int> &arr) {
        unordered_map<int, int> m;
        for (auto &num : arr) {
            ++m[num];
        }
        for (auto &[k, v] : m) {
            if (k == 0 && v > 1) return true;
            if (k != 0 && m.count(k * 2)) return true;
        }
        return false;
    }
};